test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_core_test.clj

```diff
@@ -301,3 +315,103 @@
+                           :request-header-max-size 100
+                           :so-linger-milliseconds 41
+                           :idle-timeout-milliseconds 9001}
+                    :https {:port 92
+                            :host "0.0.0.0"
+                            :protocols nil
+                            :cipher-suites nil
+                            :keystore-config
+                            {:truststore (-> (KeyStore/getDefaultType)
+                                             (KeyStore/getInstance))
+                             :key-password "hello"
+                             :keystore (-> (KeyStore/getDefaultType)
+                                           (KeyStore/getInstance))}
+                            :request-header-max-size 100
+                            :so-linger-milliseconds 42
+                            :idle-timeout-milliseconds 9002
+                            :client-auth :want}
+                    :jmx-enable false
+                    :max-threads 100
+                    :queue-max-size 101})
+          connectors (.getConnectors server)]
+      (is (= 2 (count connectors))
+          "Unexpected number of connectors for server")
+      (is (= 25 (.getPort (first connectors)))
+          "Unexpected port for first connector")
+      (is (= 9001 (.getIdleTimeout (first connectors)))
+          "Unexpected idle timeout for first connector")
+      (is (= 92 (.getPort (second connectors)))
+          "Unexpected port for second connector")
+      (is (= 9002 (.getIdleTimeout (second connectors)))
+          "Unexpected idle time for second connector"))))
+
+(deftest test-idle-timeout
+  (let [read-lines (fn [r]
+                     (let [sb (StringBuilder.)]
+                       (loop [l (.readLine r)]
+                         (when l
+                           (.append sb l)
+                           (.append sb "\n")
+                           ;; readLine will block until the socket is closed,
+                           ;; or will throw a SocketTimeoutException if there
+                           ;; is no data available within the SoTimeout value.
+                           (recur (.readLine r))))
+                       (.toString sb)))
+        body "Hi World\n"
+        path "/hi_world"
+        ring-handler (fn [req] {:status 200 :body body})
+        read-response (fn [client-so-timeout]
+                        (let [s (Socket. "localhost" 9000)
+                              out (PrintWriter. (.getOutputStream s) true)]
+                          (.setSoTimeout s client-so-timeout)
+                          (.println out (str "GET " path " HTTP/1.1\n"
+                                             "Host: localhost\n"
+                                             "\n"))
+                          (let [in (BufferedReader. (InputStreamReader. (.getInputStream s)))]
+                            (read-lines in))))]
+    (let [config {:webserver {:port 9000
```

Comment:
Would it be worth combining this `let` into the previous one? It seems weird to me to have two nested `let` calls one after the other.